src/EditProvider.ts

```diff
@@ -105,14 +107,42 @@ export class EditProvider
   }
 
   public static beautifyOptions(
-    filePath: string = vscode.workspace.rootPath
+    path: string = vscode.workspace.rootPath
   ): Promise<LanguageOptionValues> {
     try {
-      const explorer = cosmiconfig("unibeautify", {});
+      const vscodeSettings = vscode.workspace.getConfiguration("unibeautify");
+      let defaultConfigOverride = vscodeSettings.defaultConfig;
+      let pathToConfig = path;
+
+      let cosmiOptions: ExplorerOptions = {};
+      const explorer = cosmiconfig("unibeautify", cosmiOptions);
       const defaultConfig: LanguageOptionValues = {};
+
+      if (defaultConfigOverride) {
+        pathToConfig = defaultConfigOverride;
+
+        return explorer
+          .load(pathToConfig)
+          .then(returnResult)
+          .catch(catchError);
+      }
+
       return explorer
-        .search(filePath)
-        .then(result => (result ? result.config : defaultConfig));
+        .search(pathToConfig)
+        .then(returnResult)
+        .catch(catchError);
+
+      function returnResult(result: any) {
+        if (!result) return defaultConfig;
+
+        return result.config;
+      }
+
+      function catchError() {
+        return vscode.window.showErrorMessage(
+          `We could not find your configuration file: ${pathToConfig}. Please correct your path, otherwise the plugin will not work!`
+        );
```

Comment:
I'd prefer removing `pathToConfig` entirely and instead passing the various paths as an argument to generate a custom error message.

Consider the following:

```typescript
if (defaultConfigOverride) {
        return explorer
          .load(defaultConfigOverride)
          .then(returnResult)
          .catch(catchError(defaultConfigOverride));
}

return explorer
        .search(filePath)
        .then(returnResult)
        .catch(catchError(filePath));

function catchError(configPath) {
        return () => vscode.window.showErrorMessage(
          `We could not find your configuration file: ${configPath}. Please correct your path, otherwise the plugin will not work!`
        );
}
```

Reply:
On second glance, I see the `catchError` message says `configuration file` whereas the second call is actually searching for config file (i.e. `explorer.search(filePath)` ) not loading the file (i.e. `explorer.load(defaultconfigOverride)`) so the error message does not make sense in both cases.

I think something like this is more appropriate:

```typescript
if (defaultConfigOverride) {
        return explorer
          .load(defaultConfigOverride)
          .then(returnResult)
          .catch((error) => {
            vscode.window.showErrorMessage(`We could not find your configuration file: ${defaultConfigOverride}. Please correct your path, otherwise the plugin will not work!`);
             throw error;
        });
}

return explorer
        .search(filePath)
        .then(returnResult);
```